concordium-consensus/src/Concordium/GlobalState/Persistent/BlobStore.hs

```diff
@@ -710,8 +710,8 @@ instance MonadBlobStore m => BlobStorable m (CachedBlobbed BlobRef f)
 --
 -- It can contain either a CachedBlobbed value or both a Blobbed value and the recursive type.
 data BufferedBlobbed ref f
-    = LBMemory (IORef (Blobbed ref f)) (f (BufferedBlobbed ref f))
-    | LBCached (CachedBlobbed ref f)
+    = LBMemory {-# UNPACK #-} !(IORef (Blobbed ref f)) !(f (BufferedBlobbed ref f))
+    | LBCached !(CachedBlobbed ref f)
```

Comment:
I would probably prefer if we get rid of the intermediate "CachedBlobbed" since it both makes it harder to understand, and does not seem to be useful on its own.

It would also make profiling output much more useful, since using it would remove one of the silly fmaps during trie lookup/traversal that is conceptually just there to make the types work.

Reply:
And instead just have the BufferedBlobbed with 3 constructors, which would remove an indirection as well.

If this indirection is somehow important, then it should be clearly documented.

Reply:
I have removed the `Blobbed`/`CachedBlobbed`/`BufferedBlobbed` types and replaced them with a simpler `BufferedFix`. This seems to be much more performant in the lookup benchmarks.